<?xml version="1.0"?>
<launch>

    <!-- Prefix of output image published by adi safety bubble stitch node -->
    <arg name="ns_prefix" default="adi_3dtof_safety_bubble_detector_stitch" />

    <!-- The expected prefix of the input topic to adi safety bubble stitch node -->
    <arg name="ns_prefix_cam" default="cam1" />
    <arg name="camera_prefixes" default="[$(arg ns_prefix_cam)]"/>

    <group ns="$(arg ns_prefix)">
        <node name="adi_3dtof_safety_bubble_detector_stitch_host_node" pkg="adi_3dtof_safety_bubble_detector" type="adi_3dtof_safety_bubble_detector_stitch_host_node" output="screen">
            <rosparam param="camera_prefixes" subst_value="True">$(arg camera_prefixes)</rosparam> 
            <param name="enable_pointcloud_generation" value="1" />
        </node>


        <!-- Launch RViz-->
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find adi_3dtof_safety_bubble_detector)/rviz/adi_3dtof_safety_bubble_detector.rviz" required="true">
        </node>
    </group>

</launch>